rsconf/component/devbox.py

```diff
@@ -35,6 +35,12 @@ def internal_build_compile(self):
         z.run_u = jc.rsconf_db.run_u
         # Only additional config for the server is the sshd config.
         z.run_d = systemd.custom_unit_prepare(self, jc, watch_files=[z.host.ssh_d])
+        u = jc.devbox.users[self.user_name]
+        self._docker_image = jc.devbox.docker_image
+        z.ssh_port = u
+        if isinstance(u, dict):
+            self._docker_image = u.docker_image
```

Comment:
Don't assume anything exists in the dict. Also testing for dict is problematic with dot form (u.docker_image). Test for PKDict. Always use get. There's self and z. I think you could just rely on z as follows:

```py
if isinstance(u, PKDict):
    z.docker_image = u.get('docker_image', z.docker_image)
    z.ssh_port = u.ssh_port
else:
    z.ssh_port  = u
```

Another point to make is that bop uses self.name to get configuration that's specific, e.g. jc.petshop.port. You could consider just looking for `devbox_<user>.docker_image`. I'm not sure that's better, but it would be consistent with bop, which does this similarly.

Reply:
2d2b6a6